client/src/pages/SignIn.js

```diff
@@ -64,12 +65,41 @@ const signInPageStyle = (theme) => ({
   },
 });
 
+
 const SignInPage = (props) => {
   const { classes } = props;
+  let history = useHistory();
+  const { user, setUser } = useContext(UserContext);
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
+  const [message, setMessage] = useState(null);
+
+  const handleSignIn = (e) => {
+    e.preventDefault();
+    const data = {email: email, password: password};
+    fetch('/auth/login', {
+      method: "POST",
+      headers: {
+        "Content-type": "application/json"
+      },
+      redirect: "manual",
+      body: JSON.stringify(data)
+    }).then(response => response.json()).then(data => {
+      if (data.status === 200) {
+        console.log(user);
+        setUser(data.user);
```

Comment:
We are going to have to think about how we persist the user being signed in, as well as how to securely manage it so that if the user refreshes the page, we make sure they don't stay logged in if their token as expired/is invalid for some reason. Let me know if you have any questions and this could be something to bring up with your team lead as well.

Reply:
With the tokens, I've got an authenticate decoratotr I made that checks if a user is logged in before allowing access to the view and it also checks if the jwt is valid. there's also a refresh token view but honestly, I'm not sure how that works since this is the first time I've ever used it.
I'm actually trying to write something to keep the returned data in localStorage. I also have an API end point that returns all the data for the user but I'm not sure how to call that, maybe check if the profile is owned by the user then get the information to fill it out?

Reply:
So, I ended up using this method... I made a function to call the get_user_info api endpoint if the user variable in UserContext is empty.  To make the page wait/re-render when it gets filled (because of the async fetch) I put a simple if statement that says ```if (user === "") {return <span>Loading...</span>``` and when the user gets filled it moved to the second if statement that says ```if (user === "redirect") {return <Redirect to="/signin" />}```
This works very well but it's not as clean as I'd like it to be so I'm doing research into modifying the UserProvider to see if there's a way I can make it fill the user variable automatically if it's not already filled.

I can push my changes but, like I said, I don't think it's a very clean solution. Let me know if I should push with this current method then update it as I find more information and making it work without having to deal with the if statements or if I should just keep trying to figure out how to make it do everything automatically before a page load (no idea how to do that so it may take alot of research, just a forewarning)